system-design/youtube-start-seconds.md

```diff
@@ -0,0 +1,58 @@
+# YouTube Iframe API - startSecond
+
+The property **startSeconds** lets you skip the given seconds from the start of video.
+
+## Brief Overview
+
+- **startSeconds** load video and skip the given seconds from the start.
+- **startSeconds** only available when the player is loaded.
+- **startSeconds** can be called at the initial load of the video.
+- Similar to **startSeconds** we have **endSeconds** for deciding when to stop the video.
+
+## How it can help with Sync
+
+We can use this configuration for people that join the group late after video has already started to make them sync to other Users.
+So when User join the host Party we will assign host youtube player current timestamp by method ytplayer.getCurrentTime(); to the start seconds i.e startSeconds = hostPlayer.getCurrentTime().
+
+##### Note: getCurrentTime is not a function needs to be created manually [This is how you can do that](https://stackoverflow.com/questions/58870093/get-current-timestamp-of-embedded-youtube-playlist-videos-from-the-iframe-tag).
+
+## Reference/Examples:
+Below flow chart explain how will sync work for user joining our app late.
+
+![alternative text](images/start-second-sync-flow.png)
```

Comment:
This contains no timing information, or the parameters being passed around, please be really specific.

- How are we accounting for the time it takes the player to load?
- How do we select the client to call `getCurrentTime`?
- Do we call `getCurrentTime` on all the clients and then decide on one value, if yes how?
- What about buffering?

Reply:
I think before knowing properly how will we be making implementation of youtube syncing with sockets it cannot be explained properly how will I know exact implementation that we are going to discuss this depends on our discussion of implementation first.

Reply:
This part could be added after having the discussion of coding implementation for this because I am not sure what more to add.

Reply:
We are using this doc for those discussions.

Document whatever approach you can come up with.

Reply:
Clarifying: you can assume anything about the sockets API

Reply:
Updated points regarding socket and added some question with solutions